osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs

```diff
@@ -210,12 +216,31 @@ public class FilterableDifficultyIcon : DifficultyIcon
         {
             private readonly BindableBool filtered = new BindableBool();
 
-            public FilterableDifficultyIcon(CarouselBeatmap item)
+            private readonly SongSelect songSelect;
+            private readonly BeatmapInfo info;
+
+            public FilterableDifficultyIcon(CarouselBeatmap item, SongSelect songSelect)
```

Comment:
screens shouldn't be passed around like this. you could have DI'd this locally rather than in the parent (although read on – that is also not what we want).